editoast/src/views/work_schedules.rs

```diff
@@ -275,6 +295,251 @@ async fn project_path(
+    )
+)]
+async fn put_in_group(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Path(WorkScheduleGroupIdParam { id: group_id }): Path<WorkScheduleGroupIdParam>,
+    Json(work_schedules): Json<Vec<WorkScheduleItemForm>>,
+) -> Result<Json<Vec<WorkSchedule>>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+
+    conn.transaction(|conn| {
+        Box::pin(async move {
+            // Check that the group exists
+            WorkScheduleGroup::retrieve_or_fail(&mut conn.clone(), group_id, || {
+                WorkScheduleError::WorkScheduleGroupNotFound { id: group_id }
+            })
+            .await?;
+
+            // Create work schedules
+            let work_schedules_changesets = work_schedules
+                .into_iter()
+                .map(|work_schedule| work_schedule.into_work_schedule_changeset(group_id))
+                .collect::<Vec<_>>();
+            let work_schedules =
+                WorkSchedule::create_batch(&mut conn.clone(), work_schedules_changesets).await?;
+
+            Ok(Json(work_schedules))
+        })
+    })
+    .await
+}
+
+#[derive(Serialize, ToSchema)]
+#[cfg_attr(test, derive(Deserialize))]
+struct GroupContentResponse {
+    #[schema(value_type = Vec<WorkSchedule>)]
+    results: Vec<WorkSchedule>,
+    #[serde(flatten)]
+    stats: PaginationStats,
+}
+
+#[derive(Debug, Clone, serde::Deserialize, utoipa::IntoParams)]
+#[into_params(parameter_in = Query)]
+pub struct WorkScheduleOrderingParam {
+    #[serde(default)]
+    pub ordering: Ordering,
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "work_schedules",
+    params(PaginationQueryParam, WorkScheduleGroupIdParam, WorkScheduleOrderingParam),
+    responses(
+        (status = 200, description = "The work schedules in the group", body = inline(GroupContentResponse)),
+        (status = 404, description = "Work schedule group not found"),
+    )
+)]
+async fn get_group(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Path(WorkScheduleGroupIdParam { id: group_id }): Path<WorkScheduleGroupIdParam>,
+    Query(pagination_params): Query<PaginationQueryParam>,
+    Query(ordering_params): Query<WorkScheduleOrderingParam>,
+) -> Result<Json<GroupContentResponse>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let ordering = ordering_params.ordering;
+    let settings = pagination_params
+        .validate(1000)?
+        .warn_page_size(100)
```

Comment:
```suggestion
        .validate(100)
```

Since this is a new endpoint I'd suggest to enforce pagination directly. The `validate/warn` exists for some routes that used to not be paginated and for which the frontend expects all the data to be returned at once (like for the rolling stock selector for example).